pkg/reconciler/managed/reconciler.go

```diff
@@ -510,6 +511,19 @@ func NewReconciler(m manager.Manager, of resource.ManagedKind, o ...ReconcilerOp
 	return r
 }
 
+// maybeBackoff updates the managed resource status and returns an error in the
+// case that backoff is enabled and the passed error is not nil. Otherwise, the
+// request is requeued with specified wait time, and the error result of the
+// update operation is returned. If the update operation error result is not
+// nil, backoff will also be triggered implicitly.
+func (r *Reconciler) maybeBackoff(ctx context.Context, m resource.Managed, wait time.Duration, err error) (reconcile.Result, error) {
+	updateErr := errors.Wrap(r.client.Status().Update(ctx, m), errUpdateManagedStatus)
+	if r.useBackoff && err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{RequeueAfter: wait}, updateErr
+}
+
```

Comment:
I'm a little averse to wrappers that produce return values like this, but I can't think of a better approach if we want to allow folks to pick between backoff or fixed requeuafter values.

Do you think we could just straight-up switch to using backoff? i.e. If/when folks bump to this version of runtime they'd be using backoff? Providers could cut over at their own leisure.

Reply:
I would feel pretty good about just switching over if we include a default implementation (and potentially some variations) that are recommended because we are essentially just using an extremely naive rate limiting function right now as it is. Would just need to explicitly call out in release notes though or else consumers would be defaulting to backing off to ~16 minutes with the controller-runtime implementation. The long wait would remain 1 minute thought so likely not too big of a problem.

Reply:
> If/when folks bump to this version of runtime they'd be using backoff? Providers could cut over at their own leisure.

That sounds good to me.

> I would feel pretty good about just switching over if we include a default implementation (and potentially some variations) that are recommended because we are essentially just using an extremely naive rate limiting function right now as it is.

I agree that a simple default rate limiter would go a long way here. I believe the big three providers would just opt for using the default, maybe except provider-azure since its requests take longer than others.